### ISSUE TITLE
Link color is not adjusted to ensure proper contrast when using the lighter text color utilities

### ISSUE THREAD

**Issue by stphnwlkr** [uswds/uswds#2632]
<img width="838" alt="link_color" src="https://user-images.githubusercontent.com/1405187/43285740-eb638680-90ed-11e8-84c0-3e36765ceff2.png">
Using the following example:

`<div class="card-lg:grid-col bg-primary-darker text-base-lightest radius-lg">
          <h2>card-lg:grid-col bg-primary-darker text-base-lightest radius-lg</h2>
          <p>Ex aliqua irure adipisicing aute velit ad culpa proident anim nisi. Et enim quis et laborum sit pariatur minim est nisi laboris fugiat aute laborum consectetur. Deserunt dolor laboris proident culpa irure aute magna aliquip  pariatur elit laboris ea. Pariatur Lorem ut deserunt cillum commodo cillum qui nisi elit amet occaecat. Tempor officia voluptate in cupidatat fugiat Lorem fugiat nostrud est tempor dolor est aliqua aute. <a href="https://v2.designsystem.digital.gov/">Tempor ad anim minim labore</a>.</p></div>`

The link color is not changed making it unreadable in some cases. Would it be possible to add classes to adjust the link color if a lighter text color is used?

This may be difficult because of the possible variations, but I am adding it for your consideration.

**Comment by maya** [comment]
Hi @stphnwlkr, thanks for opening this issue. 

We have this CSS from 1.0 that's also in 2.0:

```
.usa-background-dark {
  background-color: $color-gray-dark;

  p,
  span {
    color: $color-white;
  }

  a {
    color: $color-gray-lighter;

    &:hover {
      color: $color-white;
    }
  }
}
```

Would something like this work? Or would you expect it to work differently?

Maybe `background-color: $color-gray-dark;` ought to be `$color-background-dark` which you can control in your settings.

Maybe links should just be white to make them more robust to different colors.

**Comment by maya** [comment]
I also see we have very similar functionality in `components/section.scss`:
```
.usa-section-dark {
  background-color: $color-primary-darkest;
  color: $color-white;

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    color: $color-primary-alt;
  }

  p {
    color: $color-white;
  }

  a {
    color: $color-gray-lighter;

    &:hover {
      color: $color-white;
    }
  }
}
```

We probably don't want both `.usa-background-dark` and `.usa-section-dark` and should choose one.

**Comment by stphnwlkr** [comment]
I think the issue will be bigger because of the new background color utilities.  Not sure if there needs to be one for every color, but certainly withinin the useable background combinations to help reinforce color contrast requirements (light text - dark background, medium light text - medium dark background, normal). The medium combination could be unneeded or may need to be a local creation, but adding it for consideration. I also think you may need to address a:visited.

**Comment by thisisdano** [comment]
We addressed `a:visited` in `2.10.1`. This is hard to do elegantly with utilities, but `usa-dark-background` should be your go-to class for changing the color of text and links within an element. Otherwise, in the Sass, you could use `set-link-from-bg($BACKGROUND_COLOR)` to set the color of link text to either standard or reverse, depending on the value of `$BACKGROUND_COLOR`.
